flexget/plugins/list/series_list.py

```diff
@@ -0,0 +1,490 @@
+    def __len__(self, session=None):
+        return len(self._db_list(session).series)
+
+    @with_session
+    def add(self, entry, session=None):
+        # Check if this is already in the list, refresh info if so
+        db_list = self._db_list(session=session)
+        db_series = self._find_entry(entry, session=session)
+        # Just delete and re-create to refresh
+        if db_series:
+            session.delete(db_series)
+        db_series = get_db_series(entry)
+        db_list.series.append(db_series)
+        session.commit()
+        return db_series.to_entry()
+
+    @with_session
+    def discard(self, entry, session=None):
+        db_series = self._find_entry(entry, session=session)
+        if db_series:
+            log.debug('deleting series %s', db_series)
+            session.delete(db_series)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    @with_session
+    def _find_entry(self, entry, session=None):
+        """Finds `SeriesListSeries` corresponding to this entry, if it exists."""
+        for id_name in supported_ids():
+            if entry.get(id_name):
+                log.debug('trying to match series based off id %s: %s', id_name, entry[id_name])
+                res = (self._db_list(session).series.join(SeriesListSeries.ids).filter(
+                    and_(
+                        SeriesListSeriesExternalID.id_name == id_name,
+                        SeriesListSeriesExternalID.id_value == entry[id_name]))
+                       .first())
+                if res:
+                    log.debug('found series %s', res)
+                    return res
+        # Fall back to title
+        name = entry.get('series_name')
+        if not name:
+            log.verbose('no series name to match, skipping')
+            return
+        log.debug('trying to match series based of name: %s ', name)
+        res = (self._db_list(session).series.filter(SeriesListSeries.title == name.lower()).first())
+        if res:
+            log.debug('found series %s', res)
+        return res
+
+    @property
+    def immutable(self):
+        return False
+
+    @property
+    def online(self):
+        """ Set the online status of the plugin, online plugin should be treated differently in certain situations,
+        like test mode"""
+        return False
+
+    @with_session
+    def get(self, entry, session):
+        match = self._find_entry(entry=entry, session=session)
+        return match.to_entry() if match else None
+
+
+class PluginSeriesList(object):
+    schema = {'type': 'string'}
+
+    @staticmethod
+    def get_list(config):
+        return SeriesList(config)
+
+    def on_task_input(self, task, config):
+        return list(SeriesList(config))
+
+
+@event('plugin.register')
+def register_plugin():
+    plugin.register(PluginSeriesList, 'series_list', api_ver=2, groups=['list'])
+
+
+@with_session
+def get_series_lists(name=None, session=None):
```

Comment:
Might be good idea to have a class for database access and have all of these in there as static methods ..